kythe/cxx/indexer/textproto/plugins/example/testdata/plugin_test.pbtxt

```diff
@@ -0,0 +1,14 @@
+# proto-file: example.proto
+# proto-message: kythe_plugin_example.Message
+
+# The 'example' plugin adds a ref from the value to the proto descriptor for the
+# field.
+
+#- @field1 ref Field1
+#- @hello ref Field1
+field1: "hello"
```

Comment:
Include an example where the field uses concatenation and might have comments between parts, e.g.
```
field3: # Comment
  "hello"
  # "world"
  "universe"
```
(where the final value should be "hellouniverse")

Reply:
Thanks, this is a good edge case that isn't currently handled.

Reply:
This also points out a potential challenge for plugin authors needing to map between the string value ("hellouniverse" here) and the bits of the string in the input. Right now the plugin is passed a string_view which spans from the 'h' in "hello" to the 'e' at the end of "universe". It would be on the plugin author to assemble the parts into the final string value and ignore the comments.

Alternatively, I could change the plugin's `AnalyzeStringValue(absl::string_view)` to something like `AnalyzeStringValue(std::string value, std::vector<absl::string_view> ranges)`. This way the analyzer handles fully parsing/assembling the string and the plugin can see what input ranges it came from.

Reply:
Oof, yeah. It's obnoxious that there's no way to get total-bytes-read from the proto Tokenizer, only line and column. On the first line, the `column`/`column_end` members should map pretty closely excpet for tabs (https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.io.tokenizer#ColumnNumber.details).  It's like they were trying to make it obnoxious to map these back to the original text :-(

You could use a block_size of 1 in the ArrayInputStream constructor and use the ByteCount() accessor, but that would likely have pretty bad performance.  I think the common case (by far) will be a single token at (0, 0), but you should be able to scan the intervening range for newlines, then tabs to adjust the character offset.  It's [expletive] annoying, but doable.  I might be able to send a CL adding byte offset to the Token but I'm not super optimistic of that.

Something like:
```
absl::string_view SubStringAfter(absl::string_view input, absl::string_view token) {
  const char* start = token.data() + token.size();
  const char* end = input.data() + input.size();
  return absl::string_view(start, end - start);
}
std::vector<absl::string_view> ReadStringTokens(absl::string_view input) {
  std::vector<absl::string_view> tokens;
  ...
  tokenizer_.set_comment_style(io::Tokenizer::SH_COMMENT_STYLE);
  while (tokenizer.Next() && tokenizer.current().type == Tokenizer::TYPE_STRING) {
    absl::string_view token = tokens.empty() ? input : SubStringAfter(input, tokens.back());
    for (int line = tokenizer.current().line; line > 0; line--) {
      token.remove_prefix(token.find('\n') + 1);
    }
    // Have to ignore the token column and look for the delimiter directly :-(
    tokens.push_back(token.substr(token.find_first_of("\"'"), tokenizer.current().text.size()));
  }
  return tokens;
}
```

Reply:
Ugh I didn't realize it'd be so hard to get byte offsets when using the tokenizer :/. Thanks for the ideas, I'll try implementing something like your code above. Ideally I'd like to find a way to use [UTF8LineIndex](https://github.com/kythe/kythe/blob/master/kythe/cxx/common/utf8_line_index.h) here, but as-is it'd be inefficient. The span we give to the tokenizer goes from the current offset to the end of the file and we wouldn't want to re-create the line index for that whole span every time. So I'd still need to find the byte offset of the end of the last piece of string field value.

Reply:
I realized that the textproto indexer (including the main branch version) wasn't handling lines with tabs correctly. I broke that out into a separate smaller PR: https://github.com/kythe/kythe/pull/4656.

Reply:
Ok, I think I've got this working now (I merged #4656 into this one).